xen/include/xen/pci.h

```diff
@@ -132,13 +132,29 @@ struct pci_dev {
 
     /* Data for vPCI. */
     struct vpci *vpci;
+#ifdef CONFIG_ARM
+    /* Shows that device is protected by IOMMU */
+    bool is_protected;
+#endif
 };
 
 #define for_each_pdev(domain, pdev) \
     list_for_each_entry(pdev, &(domain)->pdev_list, domain_list)
 
 #define has_arch_pdevs(d) (!list_empty(&(d)->pdev_list))
 
+#ifdef CONFIG_ARM
+static inline void pci_device_set_protected(struct pci_dev *pdev)
```

Comment:
Is it possible that x86 can also implement something like this always returning 'true'?
So, we don't have ifdef ARM?